ci/build.py

```diff
@@ -117,6 +123,8 @@ def run_cmd():
     image_id = _get_local_image_id(docker_tag=tag)
     if not image_id:
         raise FileNotFoundError('Unable to find docker image id matching with {}'.format(tag))
+    # now that we've built the container, push it to our docker cache
+    push_docker_cache(registry, tag, image_id)
```

Comment:
This would be executed on developer machines as well. Is it necessary to trigger the push here? Currently we have a separate `docker-compose push` call in the Jenkins pipeline

Reply:
Good point. Right now, it will fail with just a warning and continue on, but maybe we should only attempt pushing the image if an environment variable (such as DOCKER_ECR_REGISTRY) is set.

Reply:
But the original point in having the push there is to have all slave machines push up an image if they build it, so we don't rely on a scheduled job to build and push these images up. With the "crowd sourced" method, the first CI slave that has to build the container will push it up, and all immediate future invocations using the same image will benefit immediately.

Reply:
How is it related to the `cache_from` entry in docker-compose? I think currently the new container build would only consider cached entries specified in `cache_from`, which would refer to the `latest` tag of the image in ECR?

Reply:
The docker image naming scheme introduced in this PR wouldn't work for what is in master. But we don't have caching issues in master (the problem is only in other branches since the docker containers share names but have different dockerfiles.)

We still should be able to move master over to using ECR at some point, just to reduce the chance of hitting dockerhub's rate limiting. For public users, we could still publish to dockerhub but shift all CI pipelines to use images from ECR.